src/pybind/mgr/balancer/module.py

```diff
@@ -1001,6 +1057,71 @@ def optimize(self, plan: Plan) -> Tuple[int, str]:
                 self.log.info(detail)
                 return -errno.EINVAL, detail
 
+    def do_read_balancing(self, plan: Plan) -> Tuple[int, str]:
+        self.log.info('do_read_balancing')
+        osdmap_dump = plan.osdmap_dump
+
+        if len(plan.pools):
+            pools = plan.pools
+        else:  # all
+            pools = [str(i['pool_name']) for i in osdmap_dump.get('pools', [])]
+        if len(pools) == 0:
+            detail = 'No pools available'
+            self.log.info(detail)
+            return -errno.ENOENT, detail
+        self.log.info('pools %s' % pools)
+
+        adjusted_pools = []
+        inc = plan.inc
+        total_num_changes = 0
+        pools_with_pg_merge = []
+        crush_rule_by_pool_name = {}
+        ec_pools = []
+        replicated_pools_with_optimal_score = []
+        small_pools = []
+        for p in osdmap_dump.get('pools', []):
+            if p['pg_num'] > p['pg_num_target']:
+                pools_with_pg_merge.append(p['pool_name'])
+            crush_rule_by_pool_name[p['pool_name']] = p['crush_rule']
+            if p['type'] == 3:
+                ec_pools.append(p['pool_name'])
+            if p['type'] == 1:
+                if p['read_balance']['score_acting'] == p['read_balance']['optimal_score']:
+                    replicated_pools_with_optimal_score.append(p['pool_name'])
+            if p['pg_num'] < 3:
```

Comment:
@JoshSalomon do you mean the number of OSDs involved with that pool (i.e. the replica number) or the total number of OSDs in the cluster?

Reply:
the mumber of OSDs involved in the pool (which is not the replica number - but you can have a pool with only HDDs and a pool with only NVMes) - the reason is that if there are way less pgs than osds in the pool it can't utilize the full cluster performance and this probably implies that the pool is probably small and insignificant. 
A second point  most chances that in this case each osd will have 0 or 1 primaries which is optimal, but in any case the point is that the load on this pool is probably insignificant or the configuration is really bad